### PR TITLE
crimson/tools/perf_staged_fltree: fix compile error

### PR DESCRIPTION
**Pull request by myoungwon**
Signed-off-by: Myoungwon Oh <myoungwon.oh@samsung.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/tools/perf_staged_fltree.cc [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include "crimson/os/seastore/onode_manager/staged-fltree/node_extent_manager.h"
 
 #include "test/crimson/seastore/onode_tree/test_value.h"
+#include "test/crimson/gtest_seastar.h"
```
Comment (cyx1231st) [inline src/crimson/tools/perf_staged_fltree.cc:16]: Looks `perf_staged_fltree` doesn't directly depends on this header, should be better to move it into `transaction_manager_test_state.h`.

Comment (cyx1231st) [inline src/crimson/tools/perf_staged_fltree.cc:16]: So we were missing that `perf_staged_fltree` also uses `EphemeralTestState` which was impacted by https://github.com/ceph/ceph/pull/53232.

So `EphemeralTestState` should tolerate that `INSTANTIATE_TEST_SUITE_P` is not defined in the non-test scenario like this, probably by defauting the j_type to "segmented" ?

**Review by cyx1231st — COMMENTED** [review]
Thanks @aravind-wdc to identify the issue!

**Review by cyx1231st — APPROVED** [review]

File: src/crimson/tools/CMakeLists.txt [hunk 2]
```diff
@@ -15,4 +15,4 @@ add_executable(perf-async-msgr perf_async_msgr.cc)
 target_link_libraries(perf-async-msgr ceph-common global ${ALLOC_LIBS})
 
 add_executable(perf-staged-fltree perf_staged_fltree.cc)
-target_link_libraries(perf-staged-fltree crimson-seastore)
+target_link_libraries(perf-staged-fltree crimson-seastore crimson::gtest)
```
Comment (aravind-wdc) [inline src/crimson/tools/CMakeLists.txt:18]: @myoungwon

I applied this patch in my environment. With it, cmake failed with below error.
(Also I had given the option -DWITH_TESTS=OFF when doing cmake.)

_====================
CMake Error at cmake/modules/BuildBoost.cmake:301 (add_executable):
Target "perf-staged-fltree" links to target "crimson::gtest" but the target
was not found. Perhaps a find_package() call is missing for an IMPORTED
target, or an ALIAS target is missing?
===============================_

I changed crimson::gtest ==> gtest, then I did not see the cmake error or the original error that I mentioned in previous post.

Comment (myoungwon) [reply src/crimson/tools/CMakeLists.txt:18]: If we disable WITH_TESTS, probably, I think gtests is also excluded from the build, right?

Comment (cyx1231st) [reply src/crimson/tools/CMakeLists.txt:18]: Hmm, probably we can specialize `EphemeralTestState` not to depend on gtest (including the inheritage to `::testing::WithParamInterface` if tests are disabled (see the usage around `UNIT_TESTS_BUILT`).

Comment (myoungwon) [reply src/crimson/tools/CMakeLists.txt:19]: @cyx1231st @aravind-wdc Updated.

**Review by cyx1231st — APPROVED** [review]
LGTM, does it work for you @aravind-wdc ?

**Comment by aravind-wdc** [comment]
> LGTM, does it work for you @aravind-wdc ?

Yes. Works for me. Thank you.
I tested with -DWITH_TESTS=OFF option only...

**Pull request merged**